src/mixins/AddSendRequestByMessage.ts

```diff
@@ -665,6 +670,270 @@ export function AddSendRequestByMessage<TBase extends ConnectorTUIBaseConstructo
+      const requestItem: FormFieldRequestItemJSON = {
+        "@type": "FormFieldRequestItem",
+        mustBeAccepted: true,
+        title: title,
+        settings: {
+          "@type": "DateFormFieldSettings",
+        },
+      }
+
+      return requestItem
+    }
+
+    private async createDoubleFormFieldRequestItem(title: string) {
+      const result = await prompts([
+        {
+          message: "[Optional] Enter the name of the unit of the requested double",
+          type: "text",
+          name: "unit",
+          format: (value) => (value.length > 0 ? value : undefined),
+        },
+        {
+          message: "[Optional] Enter a number as lower limit for the requested double",
+          type: "number",
+          name: "min",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+        {
+          message: "[Optional] Enter a number as upper limit for the requested double",
+          type: "number",
+          name: "max",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+      ])
+
+      const requestItem: FormFieldRequestItemJSON = {
+        "@type": "FormFieldRequestItem",
+        mustBeAccepted: true,
+        title: title,
+        settings: {
+          "@type": "DoubleFormFieldSettings",
+          unit: result.unit,
+          min: result.min,
+          max: result.max,
+        },
+      }
+
+      return requestItem
+    }
+
+    private async createIntegerFormFieldRequestItem(title: string) {
+      const result = await prompts([
+        {
+          message: "[Optional] Enter the name of the unit of the requested integer",
+          type: "text",
+          name: "unit",
+          format: (value) => (value.length > 0 ? value : undefined),
+        },
+        {
+          message: "[Optional] Enter an integer as lower limit for the requested integer",
+          type: "number",
+          name: "min",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+        {
+          message: "[Optional] Enter an integer as upper limit for the requested integer",
+          type: "number",
+          name: "max",
+          format: (value) => (typeof value === "number" ? value : undefined),
+        },
+      ])
+
+      const requestItem: FormFieldRequestItemJSON = {
+        "@type": "FormFieldRequestItem",
+        mustBeAccepted: true,
+        title: title,
+        settings: {
+          "@type": "IntegerFormFieldSettings",
+          unit: result.unit,
+          min: result.min,
+          max: result.max,
+        },
+      }
+
+      return requestItem
+    }
+
+    private async createRatingFormFieldRequestItem(title: string) {
+      const result = await prompts([
+        {
+          message: "Enter an integer between five and ten as upper limit for the requested rating",
+          type: "number",
+          name: "maxRating",
+        },
```

Comment:
also make sure that this is an integer.

Additionally, this is only one prompt, so you can remove the array.